### DISCUSSION TITLE
Transpiler Support [SCSS/TS]

### DISCUSSION THREAD

**Discussion by Panakotta00** [livebud/bud#115] [Ideas]
Can you already make use of TS and SCSS?

If not, this would be a really neat feature to have.

**Comment by matthewmueller** [comment]
Typescript should work if you import it, but it won't work inside a Svelte view yet. I believe ESBuild supports TS imports by default. We'll add Svelte + TS support when there's a renewed focus on views.

There's some [unexposed, unfinished support](https://github.com/livebud/bud/blob/f0211fc7ab9044c85cc96e040d3f725c3ff7f12e/internal/generator/transform/transform_test.go#L40-L58) for custom transforms for views, so you could have `view/index.md`. I plan to finish that up when I start migrating docs to their own website.

For SCSS, no plans currently. How would you expect to use it? Import it in JS? Reference it in a `<link>` tag? Why do you prefer SCSS over scoped, plain CSS?

**Reply by Panakotta00** [reply]
In regards of scss.
I simply prefer sass/scss over plain css especially when working with more complex components.
It reduces redundancy afterall quite a bit.

Ideal it would be used similar to how it works in svelte-kit:
```svelte
<style lang="scss">
    ...
</style>
```

It actually came to mind, ideally, PostCSS support would be great, as we would be able to use tailwind css postporcessor and their like...

**Reply by matthewmueller** [reply]
PostCSS would probably take priority for me since it's such a huge ecosystem. Sveltekit supporting scss might be a good enough reason to support it too, I'll try and make transforms powerful enough to pull this off.
